File: algorithms/easy/minimum_operations_to_make_array_values_equal_to_k.py
from typing import List

class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        N = len(nums)
        nums.sort()
        res = 0
        for i in range(N-1, -1, -1):
            if nums[i] < k:
                return -1
            if i < N - 1 and nums[i] != nums[i+1]:
                res += 1
        if nums[0] > k:
            res += 1
        return res

# Main section
for nums, k in [
                  ([5,2,5,4,5], 2),
                  ([2,1,2], 2),
                  ([9,7,5,3], 1),
               ]:
    print(f'nums, k = {nums}, {k}')
    sol = Solution()
    r = sol.minOperations(nums, k)
    print(f'r = {r}')
    print('========================')

File: algorithms/easy/minimum_operations_to_make_array_values_equal_to_k.go
package main

import "fmt"

func minOperations(nums []int, k int) int {
	freq := make(map[int]int)
	for _, n := range nums {
		freq[n]++
	}
	res := 0
	for key, _ := range freq {
		if key > k {
			res++
		}
		if key < k {
			return -1
		}
	}
	return res
}

func main() {
	type Pair struct {
		nums []int
		k    int
	}
	for _, v := range []Pair{Pair{[]int{5, 2, 5, 4, 5}, 2},
		Pair{[]int{2, 1, 2}, 2},
		Pair{[]int{9, 7, 5, 3}, 1},
	} {
		fmt.Printf("nums, k = %v, %d\n", v.nums, v.k)
		r := minOperations(v.nums, v.k)
		fmt.Printf("r = %d\n", r)
		fmt.Println("=======================")

	}
}
